lib/api/readable.js

```diff
@@ -146,15 +146,23 @@ module.exports = class BodyReadable extends Readable {
 
   async dump (opts) {
     let limit = opts && Number.isFinite(opts.limit) ? opts.limit : 262144
+    const signal = opts && opts.signal && opts.signal instanceof AbortSignal ? opts.signal : null
+    if (signal) {
+      signal.addEventListener('abort', () => {
+        this.destroy()
+      })
+    }
     try {
       for await (const chunk of this) {
```

Comment:
```suggestion
      for await (const chunk of this) {
        signal.throwIfAborted()
```

Reply:
since when it is aborted we are calling .destroy, wont this automatically error?

Reply:
depends on the timing, not guaranteed

Reply:
understood, adding these back